pkg/services/sqlstore/search_builder.go

```diff
@@ -133,12 +148,7 @@ func (sb *SearchBuilder) buildTagQuery() {
 	sb.buildSearchWhereClause()
 
 	// this ends the inner select (tag filtered part)
-	sb.sql.WriteString(`
-		GROUP BY dashboard.id HAVING COUNT(dashboard.id) >= ?
-		ORDER BY dashboard.id` + dialect.Limit(int64(sb.limit)) + `) as ids
-		INNER JOIN dashboard on ids.id = dashboard.id
-	`)
-
+	sb.sql.WriteString(`GROUP BY dashboard.id HAVING COUNT(dashboard.id) >= ? `)
```

Comment:
Maybe I'm just tired but can't for my life understand why the `GROUP BY dashboard.id HAVING COUNT(dashboard.id) >= ?` is needed? Maybe the comment can be updated to explain this?

Reply:
This is the crucial part of this part of the query, this part of the query does filtering based on tags. 

So we join on the tags (created a cartesian product), then filter based on the tags. then we want to filter all all dashboards that have same number of rows as there are tag filters. This is apperently one of the better ways to do filtering on tags in SQL (a many-to-many relationship) .

Reply:
Gotcha :+1: